decoder.go

```diff
@@ -10,12 +10,13 @@ import (
 	"io"
 	"reflect"
 	"sync"
+
+	metrics "github.com/rcrowley/go-metrics"
 )
 
-// tooBig provides a sanity check for sizes; used in several places.
-// Upper limit of 1GB, allowing room to grow a little without overflow.
-// TODO: make this adjustable?
-const tooBig = 1 << 30
+// Sanity check on incoming message sizes; This needs to be increased in some
+// places (bm CPM model snapshot)
+var DecoderMaxMsgSizeBytes = int64(1 << 30)
```

Comment:
It can't; making it an instance var means threading it through `snapshot_gen.go` in rex, I figured this would save time and be much cleaner. Check the receiver in `snapshot_gen.go` for the methods that use gobi :trollface: